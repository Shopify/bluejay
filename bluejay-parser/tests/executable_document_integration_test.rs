use bluejay_parser::{ast::executable::ExecutableDocument, Error};

#[test]
fn test_error() {
    insta::glob!("test_data/executable_document/error/*.graphql", |path| {
        let input = std::fs::read_to_string(path).unwrap();
        let executable_document = ExecutableDocument::parse(input.as_str());
        assert!(
            executable_document.is_err(),
            "Document did not have any errors"
        );
        let errors = executable_document.unwrap_err();
        let formatted_errors = Error::format_errors(input.as_str(), errors);
        insta::assert_snapshot!(formatted_errors);
    });
}

#[test]
fn test_valid() {
    insta::glob!("test_data/executable_document/valid/*.graphql", |path| {
        let input = std::fs::read_to_string(path).unwrap();
        let executable_document = ExecutableDocument::parse(input.as_str());
        assert!(executable_document.is_ok(), "Document had errors");
    });
}
