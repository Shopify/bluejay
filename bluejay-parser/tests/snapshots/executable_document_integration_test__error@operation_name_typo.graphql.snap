---
source: bluejay-parser/tests/executable_document_integration_test.rs
expression: formatted_errors
input_file: bluejay-parser/tests/test_data/executable_document/error/operation_name_typo.graphql
---
Error: Unexpected token
   ╭─[<unknown>:1:1]
   │
 1 │ querry {
   │ ───┬──  
   │    ╰──── Unexpected token
───╯

